### PR TITLE
Use JUnit 5 in `base` module

### PR DESCRIPTION
**Pull request by dmytro-grankin**
This PR removes usage of JUnit 4 in `base` module.

Usages of JUnit 4 are still present in `tools`. They will be removed within a separate PR.

### CODE REVIEW

**Comment by dmytro-grankin** [comment]
@armiol PTAL.

File: base/src/test/java/io/spine/code/proto/TypeSetTest.java [hunk 1]
```diff
@@ -21,24 +21,29 @@
...
         FileDescriptor file = fileSet.tryFind(FileName.of("google/protobuf/descriptor.proto"))
                                      .get();
         TypeSet typeSet = TypeSet.messagesAndEnums(file);
         assertFalse(typeSet.isEmpty());
     }
 
     @Test
-    public void obtain_messages_and_enum_from_a_set() {
+    @DisplayName("obtain message and enums")
+    void obtain_messages_and_enum_from_a_set() {
```
Comment (dmdashenkov) [inline base/src/test/java/io/spine/code/proto/TypeSetTest.java]: Please migrate to the proper camel case.

File: base/src/test/java/io/spine/json/JsonTest.java [hunk 2]
```diff
@@ -97,7 +98,8 @@ public void print_to_compact_json() {
     }
 
     @Test
-    public void parse_from_json() {
+    @DisplayName("parse from JSON")
+    void parse_from_json() {
```
Comment (dmdashenkov) [inline base/src/test/java/io/spine/json/JsonTest.java]: Same as above.

File: base/src/test/java/io/spine/protobuf/AnyPackerTest.java [hunk 3]
```diff
@@ -55,26 +57,26 @@
...
-        assertEquals(Any.pack(spineMsg).getValue(), actual.getValue());
+        assertEquals(Any.pack(spineMsg)
+                        .getValue(), actual.getValue());
         assertEquals(typeUrl.value(), actual.getTypeUrl());
     }
 
     @Test
-    public void unpack_spine_message_from_Any() {
+    @DisplayName("unpack Spine message from Any")
+    void unpack_spine_message_from_Any() {
```
Comment (dmdashenkov) [inline base/src/test/java/io/spine/protobuf/AnyPackerTest.java]: Same as above.

File: base/src/test/java/io/spine/protobuf/MessageFieldExceptionTest.java [hunk 4]
```diff
@@ -25,17 +25,20 @@
...
 
-@SuppressWarnings("ThrowableNotThrown")
-public class MessageFieldExceptionShould {
+@DisplayName("MessageFieldException should")
+class MessageFieldExceptionTest {
 
     @Test
-    public void construct_instance_with_formatted_message() {
+    @DisplayName("construct an instance with a formatted message")
+    void construct_instance_with_formatted_message() {
```
Comment (dmdashenkov) [inline base/src/test/java/io/spine/protobuf/MessageFieldExceptionTest.java]: Same as above.

**Comment by dmytro-grankin** [comment]
The change of inspection profiles is a part of the [issue](https://github.com/SpineEventEngine/config/issues/22) in `config`. And also I have created #234 to fix the test methods naming.

**Pull request merged**